From f6d719cab065bd53e8b958930b99bec903d591c0 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 24 Jul 2017 13:59:25 +0800
Subject: [PATCH 2211/5242] MLK-16075-16 gpu: imx: dpu: fetchdecode: Add
 helper fetchdecode_get_fetcheco()

commit  11e86b505685e0c16e35fa614af30af221352993 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds helper fetchdecode_get_fetcheco() so that users may
get the relevant fetcheco via fetchdecode.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-fetchdecode.c |   20 ++++++++++++++++++++
 include/video/dpu.h                   |    1 +
 2 files changed, 21 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-fetchdecode.c b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
index df1f454..bf5bc0c 100644
--- a/drivers/gpu/imx/dpu/dpu-fetchdecode.c
+++ b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
@@ -503,6 +503,26 @@ u32 fetchdecode_get_vproc_mask(struct dpu_fetchdecode *fd)
 }
 EXPORT_SYMBOL_GPL(fetchdecode_get_vproc_mask);
 
+struct dpu_fetcheco *fetchdecode_get_fetcheco(struct dpu_fetchdecode *fd)
+{
+	struct dpu_soc *dpu = fd->dpu;
+
+	switch (fd->id) {
+	case 0:
+	case 1:
+		return dpu->fe_priv[fd->id];
+	case 2:
+	case 3:
+		/* TODO: for DPU v1, add FetchEco2 support */
+		return dpu->fe_priv[fd->id - 2];
+	default:
+		WARN_ON(1);
+	}
+
+	return ERR_PTR(-EINVAL);
+}
+EXPORT_SYMBOL_GPL(fetchdecode_get_fetcheco);
+
 struct dpu_hscaler *fetchdecode_get_hscaler(struct dpu_fetchdecode *fd)
 {
 	struct dpu_soc *dpu = fd->dpu;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 2f36e79..d1c0949 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -631,6 +631,7 @@ void layerblend_pixengcfg_clken(struct dpu_layerblend *lb,
 struct dpu_vscaler *dpu_vs_get(struct dpu_soc *dpu, int id);
 void dpu_vs_put(struct dpu_vscaler *vs);
 
+struct dpu_fetcheco *fetchdecode_get_fetcheco(struct dpu_fetchdecode *fd);
 struct dpu_hscaler *fetchdecode_get_hscaler(struct dpu_fetchdecode *fd);
 struct dpu_vscaler *fetchdecode_get_vscaler(struct dpu_fetchdecode *fd);
 
-- 
1.7.9.5

